### PR TITLE
fix connectors build for upstream `estuary/flow` changes

### PR DESCRIPTION
**Pull request by jgraettinger**
This follows-up from estuary/flow#289 to fix the connectors build.

### CODE REVIEW

**Review by saterus — APPROVED** [review]
This looks good to me.

File: tests/run.sh [hunk 1]
```diff
@@ -19,55 +21,65 @@ function bail() {
...
+# --tempdir to use our known TESTDIR rather than creating a new temporary directory.
+# --unix-sockets to create UDS socket files in TESTDIR in well-known locations.
+${TESTDIR}/flowctl temp-data-plane \
+    --log.level info \
+    --poll \
+    --sigterm \
+    --tempdir ${TESTDIR} \
+    --unix-sockets \
+    &
+DATA_PLANE_PID=$!
```
[the remark this answers is not in the document]
Comment (jgraettinger) [reply tests/run.sh:77]: Sure, I re-worked it to always tear it down via trap. Confirmed it tears down the data-plane if, for example, the connector setup script fails.

File: tests/run.sh [hunk 2]
```diff
@@ -19,55 +21,65 @@ function bail() {
...
+CATALOG_SOURCE="${TESTDIR}/test.flow.yaml"
+# SQLite database into which the test is expected to materialize.
+OUTPUT_DB="${TESTDIR}/materialization.db"
+# Actual materialization output scraped from ${OUTPUT_DB}.
+ACTUAL="${TESTDIR}/actual_test_results.txt"
+
+# Ensure we start with an empty dir, since temporary data plane files will go here.
+# For now we explicitly require that the user remove it.
+if [[ -d "${TESTDIR}" ]]; then
+    bail "The test directory must not already exist prior to a test run. Consider deleting $TESTDIR if running manually"
```
[the remark this answers is not in the document]
Comment (jgraettinger) [reply tests/run.sh:40]: Yep, updated.

File: tests/run.sh [hunk 3]
```diff
@@ -19,55 +21,65 @@ function bail() {
...
 
+# Always use the latest development image to verify the mutual integration
+# of connectors and the Flow runtime. Pull to bust a cached version,
+# and copy out binaries we'll need for this test.
+FLOW_IMAGE="ghcr.io/estuary/flow:dev"
+docker pull ${FLOW_IMAGE}
+FLOW_CONTAINER=$(docker create $FLOW_IMAGE)
+docker cp ${FLOW_CONTAINER}:/usr/local/bin/flowctl ${TESTDIR}/flowctl
+docker cp ${FLOW_CONTAINER}:/usr/local/bin/etcd ${TESTDIR}/etcd
+docker cp ${FLOW_CONTAINER}:/usr/local/bin/gazette ${TESTDIR}/gazette
```
[the remark this answers is not in the document]
Comment (jgraettinger) [reply tests/run.sh:62]: None that I've seen. In fact, seeing how well this works puts some weight behind using docker images to ship releases, even if there's no intention to run _within_ a container.

**Pull request merged**